nussl/datasets/base_dataset.py

```diff
@@ -227,6 +228,19 @@ def __getitem__(self, i):
 
         return data
 
+    def __iter__(self):
+        """
+        Calls ``self.__getitem__`` from ``0`` to ``self.__len__()``.
+        Required when inheiriting Iterable.
```

Comment:
Typo in docs. Also, I assume this function doesn't mess up anything else? Using DataLoader with `num_workers > 0` is okay?

Reply:
I wrote a test, and it seems that this will work with num_workers>0. According to this thread: pytorch/pytorch#3636 (comment), I seem to be re-implementing what python would do we were just to iterate over an object with only __getitem__ implemented. (copied from deleted comment)